integration_tests/Containerfile

```diff
@@ -0,0 +1,82 @@
+# Use the base Debian image from the Podman hub
+FROM docker.io/library/debian
+
+
+
+# INSTALLING DEPENDENCIES
+
+# Update package repositories and install necessary packages
+RUN apt update -y
+
+# Copy and install dependencies before setup to save build time 
+COPY ../documentation/dependencies/debian.dependencies /dependencies/debian.dependencies
+COPY ./integration_tests/dependencies/kernel.dependencies /dependencies/kernel.dependencies
+
+RUN DEBIAN_DEPENDENCIES=$(cat /dependencies/debian.dependencies) \
+    && apt install -y $DEBIAN_DEPENDENCIES
+
+RUN KERNEL_DEPENDENCIES=$(cat /dependencies/kernel.dependencies) \
+    && apt install -y $KERNEL_DEPENDENCIES
+
+
+
+# CREATING USER
+
+# Create a new user
+RUN useradd -ms /bin/bash user
+
+# Switch to user
+USER user
+
+# Set the working directory to the user's home directory
+WORKDIR /home/user
+
+
+
+# CLONE KERNEL TREE
+
+# TODO: Is cached. What if kernel is updated?
+RUN git clone --depth 1 https://github.com/torvalds/linux.git
+
+
+
+# INSTALL LOCAL KW
+
+# Install local version of kw
+COPY ../. kworkflow 
+
+USER root
+RUN chown user:user kworkflow
+USER user
+
+RUN cd ./kworkflow && ./setup.sh -i --force
+
+# Add kw to PATH
+ENV PATH="/home/user/.local/bin:${PATH}"
+
+
+# Clone shunit2
+# RUN cd ./tests \
+#     && git clone https://github.com/kward/shunit2.git 
+
+
+
+USER root
+# Cleanup and remove unnecessary packages
+RUN apt-get autoremove -y \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+USER user
+
+
+
+# SET KERNEL AND WORKDIR
+
+WORKDIR ./linux
+# To execute commands inside kernel tree
+RUN zcat /proc/config.gz > .config
```

Comment:
The command can lead to compatibility issues when the host kernel version differs from the container kernel version. This is due to the fact that containers share the same kernel as the host system, and the container's behavior is dependent on the kernel of the host system.

For instance, running the above command within a Debian container hosted on Fedora may encounter problems. The `zcat /proc/config.gz` command may not find the `config.gz` in `/proc` because Fedora uses the `cp /boot/config-$(uname -r) .config` command to build the `.config`. Even if we manage to create the `.config` file, it may not be suitable for use in the Debian container due to differences in kernel configurations between Fedora and Debian.

## Potential Solution: Maintain Versioned .config Files

To address kernel version compatibility issues, we can consider maintaining versioned `.config` files within the project repository. This allows the selection of the appropriate `.config` file that matches the distribution of the container, ensuring compatibility.

I'm not sure if this is the best solution so feel free to correct me or add new informations!

Reply:
From our discussion, we'll not deploy a new kernel in the container; for that, we'll use Vagrant.